notes_app/notes/tests.py

```diff
@@ -27,79 +29,91 @@ def get_credentials(self):
+            "resource_uri": self.api_detail,
             "title": "Hello world!",
             "updated_on": "2016-09-09T08:02:46.504000"
         })
 
     def test_get_detail_unauthenticated(self):
+        """Access to a note should be denied to unauthenticated users."""
         self.assertHttpUnauthorized(self.api_client.get(
-            '/api/v1/node/1/', format='json'))
+            self.api_detail, format='json'))
 
     def test_get_detail_json(self):
+        """Authenticated user should have access to a note they own."""
         resp = self.api_client.get(
-            '/api/v1/note/1/',
+            self.api_detail,
             format='json',
             authentication=self.get_credentials())
         self.assertValidJSONResponse(resp)
         self.assertKeys(self.deserialize(resp), [
-            'title', 'content', 'created_on', 'updated_on'])
-        self.assertEqual(self.deserialize(resp)['title'], 'Hello world!')
+            u'id', u'title', u'content', u'created_on', u'updated_on',
+            u'resource_uri'])
+        self.assertEqual(self.deserialize(resp)[u'title'], 'Hello world!')
 
     def test_post_list_unauthenticated(self):
+        """Unauthenticated user shouldn't be able to create notes."""
         self.assertHttpUnauthorized(self.api_client.post(
-            '/api/v1/note/', format='json', data=self.post_data))
+            self.api_list, format='json', data=self.post_data))
 
     def test_post_list(self):
+        """Authenticated user should be able to create notes."""
         self.assertEqual(Note.objects.count(), 3)
         self.assertHttpCreated(self.api_client.post(
-            '/api/v1/note/',
+            self.api_list,
             format='json',
             data=self.post_data,
             authentication=self.get_credentials()))
         self.assertEqual(Note.objects.count(), 4)
 
     def test_put_detail_unauthenticated(self):
+        """Unauthenticated user shouln't be able to update notes."""
         self.assertHttpUnauthorized(self.api_client.put(
-            '/api/v1/note/1/', format='json'))
+            self.api_detail, format='json'))
 
     def test_put_detail(self):
+        """Authenticated user should be able to update notes owned by them."""
         original_data = self.deserialize(self.api_client.get(
-            '/api/v1/note/1/', format='json',
+            self.api_detail, format='json',
             authentication=self.get_credentials()))
         new_data = original_data.copy()
         new_data['title'] = 'New Title'
         new_data['content'] = 'New Content'
         self.assertEqual(Note.objects.count(), 3)
 
         self.assertHttpAccepted(self.api_client.put(
-            '/api/v1/note/1/',
+            self.api_detail,
             format='json',
             data=new_data,
             authentication=self.get_credentials()))
-        self.assertEqual(Note.objects.count(), 4)
+        self.assertEqual(Note.objects.count(), 3)
```

Comment:
3 looks like magic number, I think it's better to get count from list endpoint and compare it to `Note.objects.count()`